### ISSUE TITLE
IDISP001 False negative in try catch

### ISSUE THREAD

**Issue by jorisvergeer** [DotNetAnalyzers/IDisposableAnalyzers#240]
I expected a IDISP001 Dispose created here.

```cs
        private void Test()
        {
            IDisposable diposable = null;
            try
            {
                diposable = new MemoryStream();
            }
            catch
            {
                diposable.Dispose();
            }
        }
```

The `diposable` is never disposed when everything is going all right.

**Comment by JohanLarsson** [comment]
Very good that you write issues with where we fail to warn, keep 'em coming!

**Comment by JohanLarsson** [comment]
Have you seen [the chat](https://gitter.im/DotNetAnalyzers/IDisposableAnalyzers)?
